### ISSUE TITLE
Deleted data suddenly reappears after restart of dgraph container

### ISSUE THREAD

**Issue by sebastian-walter** [dgraph-io/dgraph#2705]
If you suspect this could be a bug, follow the template.

- What version of Dgraph are you using?
1.0.5

- Have you tried reproducing the issue with latest release?
No

- What is the hardware spec (RAM, OS)?
We are using Google Cloud to host our Kubernetes Cluster with Dgraph running in one pod

- Steps to reproduce the issue (command/config used to run Dgraph).
We are using Dgraph inside a Kubernetes cluster. After restarting the dpragh container on our live system due to an update to our live environment. We experienced that deleted data appeared in query results. So although the node was deleted and all of the predicates are empty, the relation between two nodes still exists. 

We of course deleted this edge before. It works for new data and it also worked before (the dgraph container was running for 120 days). So, I can exclude the possibility that something on our side is broken. 

The data where the errors occur looks like this (simplified):
```
type Objective struct {
	UID                   string                `json:"uid,omitempty"`
	Title                 string                `json:"objective.title,omitempty"`
	KeyResults            []KeyResult           `json:"~keyResult.objective,omitempty"` // this is just for unmarshalling, we do not send this to dgraph
}
type KeyResult struct {
	UID             string              `json:"uid,omitempty"`
	Title           string              `json:"keyResult.title,omitempty"`
	Objective       []Objective         `json:"keyResult.objective,omitempty"` //this is being stored in dgraph
```

This is the query we are using:
```
{
  objectives(func: uid(<uid>)) {
    objective.title
    ~keyResult.objective{
        keyResult.title
        uid
     }
  }
}
```
The expected result is the following (inside of data):
```
    "objectives": [
      {
        "objective.title": "Improve the User experience for giving feedback",
        "~keyResult.objective": [
          {
            "keyResult.title": "Increase the participation rate of EE surveys to 50% on average for paying clients",
            "uid": "0xc8324"
          }
        ],
        "uid": "0x94ed7"
      }
    ]
```
but we get this:
```
    "objectives": [
      {
        "objective.title": "Improve the User experience for giving feedback",
        "~keyResult.objective": [
          {
            "uid": "<deleted-uid>" // this one was deleted before
          }
          {
            "keyResult.title": "Increase the participation rate of EE surveys to 50% on average for paying clients",
            "uid": "0xc8324"
          }
        ],
        "uid": "0x94ed7"
      }
    ]
```

We were not able to reproduce the bug locally by exporting the live data and importing it again in our local cluster. 

- Expected behaviour and actual result.
The expected behaviour is that the deleted data does not occur anymore.

**Comment by sebastian-walter** [comment]
Thanks for the quick reply. I will try that and get back to you as soon as possible, probably Thursday.

**Comment by danielmai** [comment]
Any updates on this, @sebastian-walter?

**Comment by sebastian-walter** [comment]
What we did first was importing the exported data again. This "solved" the problem with deleted UIDs popping up again. 

But that lead to another data loss. Some of the relations between our nodes where simply not there anymore. 

Concrete Example:
``` 
type Objective struct {
	UID                   string                `json:"uid,omitempty"`
	Title                 string                `json:"objective.title,omitempty"`
        Owners                []Group         `json:"objective.owners,omitempty"`
}
```
The relation of owners did not exist anymore.

We now upgraded to v.1.0.10 but so far couldn't reproduce the errors on our dev environment (we have a kubernetes setup with minikube).

We are still investigating since this leaves a very bad feeling about the decision to work with dgraph.
